src/pages/trainer/training/edit/components/toolbar/toolbar.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import {ToolbarButton} from './buttons';
+import {Icon, iconEnums} from './icons';
+import {IToolbarCommand} from './toolbarCommand';
+
+interface IProps {
+  selectToolbarCommand: (toolbarCommand: IToolbarCommand) => void;
+}
+
+export const ToolbarComponent = (props: IProps) => {
+  return (
+    <div className="btn-toolbar">
+      <div className="btn-group">
+        <ToolbarButton
+          caret="# "
```

Comment:
Now I get it, you are just inserting the markdown caret. sorry maybe I have explained my self wrong, I just wanted to get something generic / more ellaborated for the icons.

There will be commands that won't update directly the editor, but display a seconday window.

Mmm... I have to get some more time and try to understand this, I think we could pass a function / callback that would handle this, and this would after that update the reducer.

Something like (pseudocode)

```
function addBoldText() {
  // ...
}

return (
  <ToolbarButton
        onClick="addBoldText.bind(this)"
       offset={1}
     />
```` 

```` 
````